### PR TITLE
refactor(detected_object_validation): add an option for filtering and validation

### PR DESCRIPTION
**Pull request by ismetatabay**
## Description

This PR adds an option for using ```object_lanelet_filter``` or ```object_position_filter``` as an filtering option and using ```obstacle_pointcloud``` or ```occupancy_grid``` as an validating option. Moreover, at the latest version of autoware.universe, we cannot launch occupancy_grid map for validator properly, this PR fixes this issue also.

## Related links

- https://github.com/autowarefoundation/autoware.universe/issues/4250

- autoware_launch PR: https://github.com/autowarefoundation/autoware_launch/pull/479/

## Tests performed

The tests are performed in logging simulator: object_lanelet_filter, object_position_filter, obstacle_pointcloud and occupancy_grid (we need change occupancy_grid_map_method to laserscan_based_occupancy_grid_map also) mode works as expected.

## Interface changes

These PRs adds ```detected_objects_filter_method``` and ```detected_objects_validation_method``` arguments to perception component launches.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml [hunk 1]
```diff
@@ -235,6 +235,13 @@
     </include>
   </group>
 
+  <group if="$(eval &quot;'$(var objects_validation_method)'=='occupancy_grid'&quot;)">
+    <include file="$(find-pkg-share detected_object_validation)/launch/occupancy_grid_based_validator.launch.xml" if="$(var use_validator)">
```
Comment (miursh) [inline launch/tier4_perception_launch/launch/object_recognition/detection/camera_lidar_fusion_based_detection.launch.xml:239]: [Q] If this line is added, no validator is launched unless "objects_validation_method == occupancy_grid", so detection pipeline stops here.
Why do you want to add this?

File: launch/tier4_perception_launch/launch/object_recognition/detection/radar_based_detection.launch.xml [hunk 2]
```diff
@@ -20,10 +19,17 @@
...
-    <arg name="output/object" value="lanelet_filtered_objects"/>
-    <arg name="filtering_range_param" value="$(var radar_lanelet_filtering_range_param)"/>
+  <!-- Filter -->
+  <include if="$(eval &quot;'$(var objects_filter_method)'=='position_filter'&quot;)" file="$(find-pkg-share detected_object_validation)/launch/object_lanelet_filter.launch.xml">
+    <arg name="input/object" value="objects_before_filter"/>
+    <arg name="output/object" value="$(var output/objects)"/>
+    <arg name="filtering_range_param" value="$(var object_recognition_detection_object_lanelet_filter_param_path)"/>
+  </include>
+
+  <include if="$(eval &quot;'$(var objects_filter_method)'=='position_filter'&quot;)" file="$(find-pkg-share detected_object_validation)/launch/object_position_filter.launch.xml">
```
Comment (miursh) [inline launch/tier4_perception_launch/launch/object_recognition/detection/radar_based_detection.launch.xml:29]: [Q] There is another filter prepared that filters radar objects based on the distance.
What is the reason for adding this switching option of radar object filtering by position?
Besides, the input/output topic name is not matched to output of above/input of below nodes.

**Review by miursh — CHANGES_REQUESTED** [review]

File: launch/tier4_perception_launch/launch/object_recognition/detection/radar_based_detection.launch.xml (hunk 2 above)
Comment (ismetatabay) [reply launch/tier4_perception_launch/launch/object_recognition/detection/radar_based_detection.launch.xml:29]: @miursh I want to control object filter mode (lanelet or position) as well over autoware_launch. As far as I know, are the input and output topic names of the lanelet_filter and position_filter is the same? By default, lanelet_filter was used in radar based detection.

Comment (ismetatabay) [reply launch/tier4_perception_launch/launch/object_recognition/detection/radar_based_detection.launch.xml]: @miursh Thank you for your reply, I get it what you mean on radar side you are right :+1:. So, I removed option on radar side (There is only lanelet filter as in current version of autoware).

Comment (ismetatabay) [reply launch/tier4_perception_launch/launch/object_recognition/detection/radar_based_detection.launch.xml]: @miursh friendly ping

**Review by miursh — APPROVED** [review]
LGTM

**Comment by ismetatabay** [comment]
@yukkysaito Hi, could you check it? Miura-san approved it :+1:

**Review by yukkysaito — APPROVED** [review]
Thank you for your PR :+1: 
LGTM

**Pull request merged**